### PR TITLE
Implement Tag Pull

### PR DESCRIPTION
**Pull request by chrisjwelly**
Implemented a new `tagpull` command which adds on top of the items in the resume with new items of a specified tags. It can work with multiple tags: `tagpull 1 #/ frontend #/ backend`.

Separated the implementation of tagpull from redit due to slightly different expected behaviour. 

--obsolete--
Considerations:
* A separate Tag Pull command instead of combining it with `redit`
* Disallow using `redit` together with the Tag Pull.
* Tag Pull should add on top of whatever you have in the resume

Name for command: `rpull`, `pull`, `tagpull`?

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/TagPullCommand.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        List<Integer> pulledSkillIds = tagList
+                .stream()
+                .map(model::getSkillsByTag)
+                .flatMap(Collection::stream)
+                .map(Skill::getId)
+                .collect(Collectors.toList());
+
+        // Concatenate both sets of items as we are adding on top
+        List<Integer> internshipIds = Stream
```
Comment (wardetu) [inline src/main/java/seedu/address/logic/commands/TagPullCommand.java]: I think this should be prefixed with `new` -> `newInternshipIds` so it's more clear.

File: src/main/java/seedu/address/logic/commands/TagPullCommand.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                .distinct()
+                .collect(Collectors.toList());
+
+        Resume editedResume = new Resume(toEdit.getName(), toEdit.getId(), toEdit.getTags());
+        model.editResume(editedResume, internshipIds, projectIds, skillIds);
+        model.setResume(toEdit, editedResume);
+        model.setResumeToDisplay();
+        model.updateFilteredItemList(PREDICATE_SHOW_ALL_ITEMS);
+        model.commitResumeBook();
+        return new TagPullCommandResult(editedResume.toString(), "Tag Pull is successful!",
```
Comment (wardetu) [inline src/main/java/seedu/address/logic/commands/TagPullCommand.java]: Could be nice to tell user how many items are pulled ?

**Review by wardetu — COMMENTED** [review]
Looks good overall.
Usage of stream is impressive. Good code quality.

Some small remarks regarding naming and feedback.

File: src/main/java/seedu/address/logic/commands/TagPullCommand.java (hunk 1 above)
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/TagPullCommand.java]: Real nice!

File: src/main/java/seedu/address/logic/parser/ResumeEditCommandParser.java [hunk 3]
```diff
@@ -28,7 +27,7 @@
     public ResumeEditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_INTERNSHIP, PREFIX_SKILL, PREFIX_PROJECT, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_INTERNSHIP, PREFIX_SKILL, PREFIX_PROJECT);
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/parser/ResumeEditCommandParser.java:30]: Good catch

File: src/main/java/seedu/address/logic/parser/TagPullCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        Index index;
+        Set<Tag> tagList;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ResumeEditCommand.MESSAGE_USAGE),
+                    pe);
+        }
+        tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/parser/TagPullCommandParser.java:37]: Careful the case of no tags supplied

**Review by helloImHai — APPROVED** [review]
Clean and I love it 🛀🛀🛀

**Review by helloImHai — APPROVED** [review]
Clean and I love it 🛀🛀🛀
Can't find anything wrong with it lol

**Pull request merged**